spf/dnscache.go

```diff
@@ -0,0 +1,26 @@
+package spf
+
+type dnsCache map[string]map[string][]string
```

Comment:
At this time code only uses it to cache TXT records. However I future-proofed it to support many record types.  CNAME flattening is a good example of a future use.